### PR TITLE
Feature/feature crud exercise

### PR DESCRIPTION
**Pull request by AlbertPG**

### CODE REVIEW

File: app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/infopet/InfoPetExercise.java [hunk 1]
```diff
@@ -1,21 +1,452 @@
...
+    }
+
+    /**
+     * Check whether the exercise period is valid.
+     * @param exerciseDate The exercise date
+     * @param exerciseStartTime The exercise start time
+     * @param exerciseEndTime The exercise end time
+     * @return True if the period is valid
+     */
+    private static boolean checkDateTimePeriod(DateButton exerciseDate, TimeButton exerciseStartTime,
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/infopet/InfoPetExercise.java:424]: Codacy Issue found: [Method length is 16 lines (max allowed is 15).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5483188)

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Coverage increased per file
===========================
+ app/src/main/java/org/pesmypetcare/mypetcare/features/pets/Medication.java  4
+ app/src/main/java/org/pesmypetcare/mypetcare/features/pets/PetHealthInfo.java  8
         

Complexity increasing per file
==============================
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/infopet/TestTrDeleteExercise.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/activities/views/datetimebuttons/TimeButton.java  3
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/exercise/TrUpdateExercise.java  8
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/infopet/TestTrUpdateExercise.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/infopet/InfoPetExercise.java  5
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/exercise/ExerciseControllersFactory.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/exercise/TrAddExercise.java  6
- app/src/main/java/org/pesmypetcare/mypetcare/activities/views/entryview/EntryView.java  5
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/infopet/TestTrAddExercise.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/activities/views/datetimebuttons/DateButton.java  2
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/exercise/TrDeleteExercise.java  5
- app/src/main/java/org/pesmypetcare/mypetcare/activities/views/datetimebuttons/DateTimeButton.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/features/pets/Exercise.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5483188&bid=17968137)

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Coverage increased per file
===========================
+ app/src/main/java/org/pesmypetcare/mypetcare/features/pets/PetHealthInfo.java  8
         

Complexity increasing per file
==============================
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/infopet/TestTrDeleteExercise.java  1
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/infopet/TestTrUpdateExercise.java  1
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/infopet/TestTrAddExercise.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/exercise/TrAddExercise.java  6
- app/src/main/java/org/pesmypetcare/mypetcare/activities/views/datetimebuttons/DateButton.java  2
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/infopet/InfoPetExercise.java  5
- app/src/main/java/org/pesmypetcare/mypetcare/activities/views/datetimebuttons/DateTimeButton.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/features/pets/Exercise.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/exercise/ExerciseControllersFactory.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/exercise/TrUpdateExercise.java  8
- app/src/main/java/org/pesmypetcare/mypetcare/activities/views/entryview/EntryView.java  5
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/exercise/TrDeleteExercise.java  5
- app/src/main/java/org/pesmypetcare/mypetcare/activities/views/datetimebuttons/TimeButton.java  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5483188&bid=17968137)

**Review by danicm47 — COMMENTED** [review]
Comment Exercise and Pet classes functions

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Coverage increased per file
===========================
+ app/src/main/java/org/pesmypetcare/mypetcare/features/pets/PetHealthInfo.java  8
+ app/src/main/java/org/pesmypetcare/mypetcare/features/pets/Medication.java  4
         

Complexity increasing per file
==============================
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/infopet/TestTrUpdateExercise.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/exercise/TrUpdateExercise.java  8
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/exercise/TrAddExercise.java  6
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/infopet/TestTrAddExercise.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/activities/views/datetimebuttons/DateTimeButton.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/activities/views/datetimebuttons/TimeButton.java  3
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/exercise/TrDeleteExercise.java  5
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/infopet/InfoPetExercise.java  5
- app/src/main/java/org/pesmypetcare/mypetcare/features/pets/Exercise.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/exercise/ExerciseControllersFactory.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/activities/views/datetimebuttons/DateButton.java  2
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/infopet/TestTrDeleteExercise.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/activities/views/entryview/EntryView.java  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5483188&bid=17968137)

**Review by danicm47 — APPROVED** [review]

**Review by enrichp — APPROVED** [review]

**Pull request merged**